rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -128,6 +176,15 @@ rcl_logging_ret_t rcl_logging_external_initialize(
 
     auto sink = std::make_unique<spdlog::sinks::basic_file_sink_mt>(name_buffer, false);
     g_root_logger = std::make_unique<spdlog::logger>("root", std::move(sink));
+    if (!should_use_old_flushing_behavior) {
+      // in this case we should do the new thing (until config files are supported)
+      // which is to configure the logger to flush periodically and on
+      // error level messages
+      spdlog::flush_every(std::chrono::seconds(5));
+      g_root_logger->flush_on(spdlog::level::err);
```

Comment:
This seems to be working properly; I slightly modified the `talker` to print out ERROR instead of INFO, and I see this flushed to the log every second.

I made a further modification to the `talker` program to publish every 1 millisecond instead of every 1 second to see the effect of this flush on CPU time.  And it is clear that this is going to cause us more CPU and disk time; it actually shows up in the top.  That said, I think that flushing on error should be the default, so I'm good with this.